pkg/nfd-master/nfd-master.go

```diff
@@ -449,6 +451,17 @@ func verifyNodeName(cert *x509.Certificate, nodeName string) error {
 	return nil
 }
 
+func isNamespaceDenied(labelNs string, deniedNs map[string]struct{}) bool {
+	for ns := range deniedNs {
+		deniedNsWildcard := fmt.Sprintf("*.%v", ns)
+		g := glob.MustCompile(deniedNsWildcard)
+		if g.Match(labelNs) {
```

Comment:
I'm not entirely sure we need to support full-fledged globs (performance, extra external dependency), we could probably do away with something like

```golang
if strings.HasPrefix(ns, "*") {
    deniedSuffix := strings.TrimLeft(ns, "*")
    if strings.HasSuffix(labelNs, deniedSuffix) {
        return true
    }
} else if labelNs == ns {
    return true
}
return false
```

And just document that we support asterisk ("*") in front of the name to deny all sub-namespaces. WDYT?

Also, we probably want to pre-process the user input at startup to repeat same string (or glob) operations over and over again.

Reply:
Yes got it, and it's a valid point I think. I just didn't understand the last point. Can you explain it to me further, please?

Reply:
> Can you explain it to me further, please?

My point is that in the current implementation for each label we create, we run through the deny list and do `glob.MustCompile` for each deny item. That is for each time a label request arrives, for each label, for each item in deny list do compile. Even though that information is basically static, doing compile once at the startup would be enough and save a lot of cpu cycles

Reply:
@marquiz what about having working with such wildcards: `prefix.*.suffix`. The aesterik-based implementation won't work in the expected way. isn't it?

Reply:
It's just me, but I wouldn't be concerned about complex regexs or globs at the moment. Think about how domain names and IP addresses generally work, for example, you don't generally specify IP ranges like 192.168.*.3. For sub-namespaces I see a clear use case, though. More advanced globbing can be added later if it is clearly needed